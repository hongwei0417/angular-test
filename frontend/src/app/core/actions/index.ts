import * as PageActions from './page.actions';
import * as HttpActions from './http.actions';

export { PageActions, HttpActions };
