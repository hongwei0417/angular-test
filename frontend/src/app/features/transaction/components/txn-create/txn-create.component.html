<!--
    <div class="transactionSetting ">
      <h3>Transaction  Setting</h3>

      <div class="fieldContent">
        <form [formGroup]="txnForm" class="transactionBasicInfo">
          <h4>Basic Info</h4>
          <div class="itemGroup bg">
            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(TransactionName?.invalid && TransactionName?.touched)">Transaction Name</span>
                <p  *ngIf="TransactionName?.invalid && TransactionName?.touched">
                  "Transaction Name" is required
                </p>
              </label>
              <input type="text" formControlName="TransactionName">
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(APBooking?.invalid && APBooking?.touched)">AP Booking</span>
                <p  *ngIf="APBooking?.invalid && APBooking?.touched">
                  "AP Booking" is required
                </p>
              </label>

              <p-autoComplete formControlName="APBooking" [suggestions]="APBookingIds"
                (completeMethod)="filterAPBookingIds($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>


            <div class="item">
              <p>Active Flag</p>
              <p-selectButton [options]="stateOptions" formControlName="ActiveFlag"></p-selectButton>
            </div>
            <div class="item">
              <label for="">Start Time</label>
              <p-calendar formControlName="date" dateFormat="yy/mm/dd" [showTime]="true" hourFormat="24" ></p-calendar>
            </div>
            <div class="item">
              <label for="">Alarm Interval Min</label>
              <input type="number" placeholder="Numbers only" formControlName="alarmIntervalMin">
            </div>
          </div>
        </form>

        <form [formGroup]="frequencyForm" class="frequencySetting">
          <h4>Frequency Setting</h4>

          <div class="bg">
            <div class="accordionGroup">
              <div class="accordionTool">
                <div class="itemGroup">
                  <div class="item">
                    <label for="" class="fieldRequired">
                      <span>Last Tran Time</span>
                    </label>
                  </div>
                  <div class="item">
                    <label for="" class="fieldRequired">
                      <span>Cron Expression</span>
                    </label>
                  </div>
                  <div class="item">
                    <button class="txtBtn small customBtnReverse line add">Generate Cron</button>
                  </div>
                </div>
                <div class="btnGroup">
                  <button class="iconBtn small customBtn add"></button>
                </div>
              </div>

              <div class="accordionItem">
                <div class="accordionHeader" (click)="toggleAccordion($event)"
                [ngClass]="{'active': accordionStates}">
                  <div class="itemGroup">
                    <div class="item">
                      <p-calendar formControlName="LastTranTime" dateFormat="yy/mm/dd" [showTime]="true" hourFormat="24" [showSeconds]="true" (click)="$event.stopPropagation()">
                      </p-calendar>
                    </div>
                    <div class="item">
                      <input type="text" placeholder="0 5 * ? * * *" formControlName="CronExpression" (click)="$event.stopPropagation()">
                    </div>
                  </div>

                  <div class="btnGroup">
                    <button class="iconBtn small customBtnReverse line remove"></button>
                    <button class="iconBtn small customBtn copy"></button>
                  </div>
                </div>

                <div class="accordionContent">
                  <div class="itemGroup">

                    <div class="item">
                      <label for="">Time Zone ID</label>
                      <p-autoComplete formControlName="TimeZone" [suggestions]="timeZoneIds"
                        (completeMethod)="filterTimeZone($event)" [dropdown]="true" placeholder="Select">
                        <ng-template let-custId pTemplate="item">
                          <div class="custName-item">
                            <div>{{custId}}</div>
                          </div>
                        </ng-template>
                      </p-autoComplete>
                    </div>

                    <div class="item">
                      <label for="">Retry Times</label>
                      <input type="number" placeholder="Numbers only" formControlName="RetryTimes">
                    </div>

                    <div class="item">
                      <label for="">Loader Buffer Time</label>
                      <input type="number" placeholder="Numbers only" formControlName="LoaderBufferTime">
                    </div>

                    <div class="item">
                      <label for="">Back To Buffer Time</label>
                      <input type="number" placeholder="Numbers only" formControlName="BackToBufferTime">
                    </div>

                    <div class="item">
                      <label for="">Shift Back To Loader Time</label>
                      <input type="number" placeholder="Numbers only" formControlName="ShiftBackToLoaderTime">
                    </div>
                  </div>

                  <div class="itemGroup">
                    <div class="item">
                      <label for="">Start At</label>
                      <p-calendar formControlName="StartAt" dateFormat="yy/mm/dd"></p-calendar>
                    </div>

                    <div class="item">
                      <label for="">End At</label>
                      <p-calendar formControlName="EndAt" dateFormat="yy/mm/dd"></p-calendar>
                    </div>

                    <div class="item">
                      <p>Skip Over Due</p>
                      <p-selectButton [options]="stateOptionsAccordion" formControlName="SkipOverDue"></p-selectButton>
                    </div>

                    <div class="item">
                      <p>Skip All Over Due</p>
                      <p-selectButton [options]="stateOptionsAccordion" formControlName="SkipAllOverDue"></p-selectButton>
                    </div>
                  </div>


                </div>
              </div>



            </div>
          </div>
        </form>


        <form [formGroup]="moduleForm" class="moduleForm">
          <h4>Choose Module <span>( Module Name / Issue )</span></h4>

          <div class="bg">
            <div class="moduleItem">
              <div class="itemGroup">
                <div class="item">
                  <p-autoComplete formControlName="moduleName" [suggestions]="moduleName"
                    (completeMethod)="filterModuleName($event)" [dropdown]="true" >
                    <ng-template let-custId pTemplate="item">
                      <div class="custName-item">
                        <div>{{custId}}</div>
                      </div>
                    </ng-template>
                  </p-autoComplete>
                </div>

                <div class="item">
                  <p-inputSwitch formControlName="issue"></p-inputSwitch>
                </div>
              </div>

              <div class="btnGroup">
                <button class="iconBtn small customBtnReverse line remove"></button>
                <button class="iconBtn small customBtn add"></button>
              </div>
            </div>

          </div>

        </form>


        <form [formGroup]="mailGroupForm" class="mailGroupForm">
          <h4>
            <span>Choose Deliver Mail Group</span>
            <button class="txtBtn small customBtnReverse line setting">Setting</button>
          </h4>
          <div class="itemGroup bg">
            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(ownerIT?.invalid && ownerIT?.touched)">Owner IT</span>
                <p  *ngIf="ownerIT?.invalid && ownerIT?.touched">
                  "Owner IT" is required
                </p>
              </label>

              <p-autoComplete formControlName="ownerIT" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(mailTo?.invalid && mailTo?.touched)">Mail To</span>
                <p  *ngIf="mailTo?.invalid && mailTo?.touched">
                  "Mail To" is required
                </p>
              </label>

              <p-autoComplete formControlName="mailTo" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(ownerCoordinator?.invalid && ownerCoordinator?.touched)">Owner Coordinator</span>
                <p  *ngIf="ownerCoordinator?.invalid && ownerCoordinator?.touched">
                  "Owner Coordinator" is required
                </p>
              </label>

              <p-autoComplete formControlName="ownerCoordinator" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(loaderOwner?.invalid && loaderOwner?.touched)">Loader Owner</span>
                <p  *ngIf="loaderOwner?.invalid && loaderOwner?.touched">
                  "Loader Owner" is required
                </p>
              </label>

              <p-autoComplete formControlName="loaderOwner" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(mailCC?.invalid && mailCC?.touched)">Mail CC</span>
                <p  *ngIf="mailCC?.invalid && mailCC?.touched">
                  "Mail CC" is required
                </p>
              </label>

              <p-autoComplete formControlName="mailCC" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(mailBCC?.invalid && mailBCC?.touched)">Mail BCC</span>
                <p  *ngIf="mailBCC?.invalid && mailBCC?.touched">
                  "Mail BCC" is required
                </p>
              </label>

              <p-autoComplete formControlName="mailBCC" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(SAPOwner?.invalid && SAPOwner?.touched)">SAP Owner</span>
                <p  *ngIf="SAPOwner?.invalid && SAPOwner?.touched">
                  "SAP Owner" is required
                </p>
              </label>

              <p-autoComplete formControlName="SAPOwner" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>

            <div class="item">
              <label for="" class="fieldRequired">
                <span *ngIf="!(MESOwner?.invalid && MESOwner?.touched)">MES Owner</span>
                <p  *ngIf="MESOwner?.invalid && MESOwner?.touched">
                  "MES Owner" is required
                </p>
              </label>

              <p-autoComplete formControlName="MESOwner" [suggestions]="mailGroup"
                (completeMethod)="filterMailGroup($event)" [dropdown]="true" placeholder="Select">
                <ng-template let-custId pTemplate="item">
                  <div class="custName-item">
                    <div>{{custId}}</div>
                  </div>
                </ng-template>
              </p-autoComplete>
            </div>
          </div>
        </form>
      </div>



    </div>

 -->
