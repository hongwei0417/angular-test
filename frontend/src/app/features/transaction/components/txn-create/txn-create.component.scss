
@import '/src/assets/styles/variables.scss';

.title{
  border-bottom: 1px solid #000;
  padding-bottom: 10px;
}


h3{
  display: flex;
  align-items: center;
  text-align: center;
  color: $primary-medium;
  margin-bottom: 10px;
}

h3:before{
  content: '1';
  margin-right: 5px;
  width: 20px;
  height: 20px;
  display: block;
  border-radius: 50%;
  background: $primary-medium;
  color: $gray-100;
  font-size: 16px;
}

h4{
  margin-bottom: 5px;
}

input{
  background: $gray-100;
}


.transactionBasicInfo .itemGroup{
  padding: 15px;
  background: $gray-200;
}

.transactionBasicInfo .item:nth-child(-n+2){
  width: calc(50% - 35px / 2);
}


.transactionBasicInfo .item:nth-child(3){
  margin-left: 0;
}

.transactionBasicInfo .item:nth-last-child(-n+3){
  margin-top: 20px;
  width: calc(33.333333% - 70px / 3);
}


.bg{
  padding: 15px;
  background: $gray-200;
}

.accordionGroup label,
.accordionGroup p{
  margin-bottom: 0;
}

.accordionGroup label{
  width: 100%;
}

.accordionGroup{
  background: $gray-100;
  border: 1px solid #ced4da;
  box-shadow: $shadow-100;
}

.accordionGroup input,
:host ::ng-deep .accordionGroup .p-calendar .p-inputtext{
  padding: 5px 10px;
  border-radius: 5px;
}

:host ::ng-deep .accordionGroup .p-calendar .p-datepicker{
  min-width: 250px;
}

:host ::ng-deep .accordionGroup .p-datepicker table td{
  padding: 0;
}

:host ::ng-deep .p-calendar:after{
  font-size: 16px;
}


//#region accordionTool
.accordionTool{
  display: flex;
  justify-content: space-between;
  padding: 5px 0;
}

.accordionTool .itemGroup:before{ //佔位空白
  content: '';
  display: block;
  width: 76px;
}

.accordionTool .itemGroup .item:first-child{
  width: 185px;
}

.accordionTool .itemGroup .item:nth-child(2){
  width: 120px;
}

.accordionTool .btnGroup button{
  margin: 0 15px;
}
//#endregion accordionTool


//#region accordionHeader
.accordionHeader{
  display: flex;
  justify-content: space-between;
  padding: 10px 0;
  position: relative;
  background: $gray-100;
  border: 1px solid #ced4da;
  cursor: pointer;
}

.accordionHeader:before{//裝飾線
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  width: 5px;
  opacity: 0;
  background: $primary-medium;
  transition: .1s;
}

.accordionHeader .itemGroup:before{ //展開箭頭
  content: '\f054';
  font-family: "Font Awesome 5 Free";
  font-weight: 900;
  color: #888888;
  font-size: 12px;
  text-align: center;

  background: #FFF;
  border-radius: 50%;
  box-shadow: 0px 0px 2px 2px rgb(0 0 0 / 15%);
  line-height: 25px;
  width: 25px;
  margin-left: 20px;

  position: relative;
  transform: rotate(0deg);
  transition: .5s;
}

.accordionHeader .itemGroup .item:first-child{
  margin-left: 30px;
  width: 185px;
}

.accordionHeader .itemGroup .item:last-child{
  width: 120px;
}

.accordionHeader .btnGroup{
  display: flex;
  opacity: 0;
}

.accordionHeader .btnGroup button{
  margin-right: 15px;
}

//#endregion accordionHeader


//#region accordionContent
.accordionContent{
  display: flex;
  justify-content: center;
  background: $gray-100;
}

.accordionContent .itemGroup{
  display: flex;
  margin: 0 20px;
  width: calc(50% - 80px / 2);
}

@media screen and (min-width:1024px){
  .accordionContent .itemGroup{
    margin: 0 80px;
    width: calc(50% - 320px / 2);
  }
}

.accordionContent .itemGroup:nth-child(2):before{//佔位空白
  content: '';
  display: block;
  width: 100%;
  height: 24px;
  margin-bottom: 5px;
}

.accordionContent .item{
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  font-size: 13px;
}


.accordionContent input{
  width: 50% ;
}

p-selectbutton{
  width: 50%;
}

:host ::ng-deep .accordionContent .p-selectbutton .p-button{
  padding: 5px;
}

:host ::ng-deep .accordionContent .p-button-label{
  font-size: 14px;
}

:host ::ng-deep .accordionContent .p-autocomplete .p-inputtext{
  padding: 5px 10px;
  min-width: 160px;
  border-radius: 5px 0 0 5px;
}

.accordionContent input,
:host ::ng-deep .accordionContent .p-autocomplete .p-inputtext{
  font-size: 12px;
}

:host ::ng-deep .accordionContent .p-button.p-button-icon-only{
  padding: 5px;
  width: 30px;
  border-radius: 0 5px 5px 0;
}




.accordionContent .item + .item{
  margin-left: 0;
  margin-top: 5px;
}


//#endregion accordionContent





.accordionContent{
  max-height: 0;
  overflow: hidden;
  opacity: 0;
  transition: .5s;
}

.accordionHeader.active ~ .accordionContent{
  max-height: 100vh;
  overflow: visible;
  opacity: 1;
  padding: 25px 0;
}

.accordionHeader.active,
.accordionHeader:hover{
  border-color: $primary-medium;
 background: $gray-200;
}

.accordionHeader.active:before,
.accordionHeader:hover:before,
.accordionHeader.active .btnGroup,
.accordionHeader:hover .btnGroup{
  opacity: 1;
}

.accordionHeader.active .itemGroup:before{ //展開箭頭
  transform: rotate(90deg);
  color: $gray-100;
  background: $primary-medium;
}
