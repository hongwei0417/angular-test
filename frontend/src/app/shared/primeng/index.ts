// module
export * from './primeng.module';
