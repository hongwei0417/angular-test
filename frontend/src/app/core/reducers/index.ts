import {
  Action,
  ActionReducer,
  ActionReducerMap,
  createFeatureSelector,
  createSelector,
  MetaReducer,
} from '@ngrx/store';
import * as fromRouter from '@ngrx/router-store';
import { environment } from '../../../environments/environment';
import { InjectionToken } from '@angular/core';

export interface IState {
  router: fromRouter.RouterReducerState;
}

export const reducers: ActionReducerMap<IState> = {
  router: fromRouter.routerReducer,
};

export const ROOT_REDUCERS = new InjectionToken<
  ActionReducerMap<IState, Action>
>('Root reducers token', {
  factory: () => {
    return {
      router: fromRouter.routerReducer,
    };
  },
});

export const metaReducers: MetaReducer<IState>[] = !environment.production
  ? []
  : [];
