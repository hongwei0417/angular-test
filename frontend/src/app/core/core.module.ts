import { HttpHelperService } from './services/httpHelper/http-helper.service';
import { CommonModule } from '@angular/common';
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { RouterModule } from '@angular/router';
import { SharedModule } from '../shared';
import { ConfigComponent } from './components/config/config.component';
import { FooterComponent } from './components/footer/footer.component';
import { InlineProfileComponent } from './components/inline-profile/inline-profile.component';
import { MenuComponent } from './components/menu/menu.component';
import { MenuitemComponent } from './components/menuitem/menuitem.component';
import { TopbarComponent } from './components/topbar/topbar.component';
import { AppComponent } from './containers/app/app.component';
import { LayoutComponent } from './containers/layout/layout.component';
import { EnsureModuleLoadedOnceGuard } from './ensure-module-loaded-once.guard';
import { EffectsModule } from '@ngrx/effects';
import { MessageService } from 'primeng/api';
import { httpInterceptorProviders } from './interceptors';

@NgModule({
  declarations: [
    AppComponent,
    LayoutComponent,
    ConfigComponent,
    FooterComponent,
    InlineProfileComponent,
    MenuComponent,
    MenuitemComponent,
    TopbarComponent,
  ],
  imports: [CommonModule, RouterModule, SharedModule],
  exports: [AppComponent],
  providers: [HttpHelperService, MessageService, httpInterceptorProviders],
})
export class CoreModule extends EnsureModuleLoadedOnceGuard {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    super(parentModule);
  }
}
