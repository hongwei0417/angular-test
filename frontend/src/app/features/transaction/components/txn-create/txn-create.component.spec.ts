import { AutoCompleteModule } from 'primeng/autocomplete';
import { CalendarModule } from 'primeng/calendar';
import { SelectButtonModule } from 'primeng/selectbutton';
import { TableModule } from 'primeng/table';
import { MockModule } from 'ng-mocks';
import { ReactiveFormsModule } from '@angular/forms';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TxnCreateComponent } from './txn-create.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('TxnCreateComponent', () => {
  let component: TxnCreateComponent;
  let fixture: ComponentFixture<TxnCreateComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        MockModule(TableModule),
        SelectButtonModule,
        CalendarModule,
        AutoCompleteModule,
        NoopAnimationsModule,
      ],
      declarations: [TxnCreateComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TxnCreateComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
