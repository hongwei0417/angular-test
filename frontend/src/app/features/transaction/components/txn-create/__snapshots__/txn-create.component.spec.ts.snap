// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxnCreateComponent should match snapshot 1`] = `
<app-txn-create
  APBookingIds={[Function Array]}
  __ngContext__={[Function LRootView]}
  accordionStates="false"
  clearState={[Function EventEmitter_]}
  errorMessage=""
  formBuilder={[Function FormBuilder]}
  frequencyForm={[Function FormGroup]}
  stateOptions={[Function Array]}
  stateOptionsAccordion={[Function Array]}
  submitted={[Function EventEmitter_]}
  timeZoneIds={[Function Array]}
  txnForm={[Function FormGroup]}
>
  <div
    class="transactionCreate"
  >
    <div
      class="container"
    >
      <div
        class="title"
      >
        <h2>
          Create Transaction
        </h2>
        <button
          class="txtBtn customBtn arrow"
          routerlink="/transaction/list"
        >
           Back 
        </button>
      </div>
      <div
        class="transactionSetting"
      >
        <h3>
          Transaction Setting
        </h3>
        <div
          class="fieldContent"
        >
          <form
            class="transactionBasicInfo ng-untouched ng-pristine ng-invalid"
            novalidate=""
          >
            <h4>
              Basic Info
            </h4>
            <div
              class="itemGroup"
            >
              <div
                class="item"
              >
                <label
                  class="fieldRequired"
                >
                  <span>
                    Transaction Name
                  </span>
                </label>
                <input
                  class="ng-untouched ng-pristine ng-invalid"
                  formcontrolname="TransactionName"
                  type="text"
                />
              </div>
              <div
                class="item"
              >
                <label
                  class="fieldRequired"
                >
                  <span>
                    AP Booking
                  </span>
                </label>
                <p-autocomplete
                  class="ng-untouched ng-pristine ng-invalid"
                  formcontrolname="APBooking"
                  placeholder="Select"
                >
                  <span
                    class="p-autocomplete p-component p-autocomplete-dd"
                  >
                    <input
                      aria-activedescendant="p-highlighted-option"
                      aria-autocomplete="list"
                      aria-controls="pr_id_1_list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      autocomplete="off"
                      class="p-autocomplete-input p-inputtext p-component p-autocomplete-dd-input"
                      placeholder="Select"
                      role="searchbox"
                      type="text"
                    />
                    <button
                      class="p-autocomplete-dropdown p-ripple p-button p-component p-button-icon-only"
                      pbutton=""
                      pripple=""
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                        class="p-button-icon pi pi-chevron-down"
                      />
                      <span
                        aria-hidden="true"
                        class="p-button-label"
                      >
                      </span>
                    </button>
                  </span>
                </p-autocomplete>
              </div>
              <div
                class="item"
              >
                <p>
                  Active Flag
                </p>
                <p-selectbutton
                  class="ng-untouched ng-pristine ng-valid"
                  formcontrolname="ActiveFlag"
                >
                  <div
                    class="p-selectbutton p-buttonset p-component"
                    role="group"
                  >
                    <div
                      aria-label="Yes"
                      aria-labelledby="Yes"
                      aria-pressed="false"
                      class="p-button p-component p-ripple"
                      pripple=""
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="p-button-label"
                      >
                        Yes
                      </span>
                    </div>
                    <div
                      aria-label="No"
                      aria-labelledby="No"
                      aria-pressed="true"
                      class="p-button p-component p-highlight p-ripple"
                      pripple=""
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="p-button-label"
                      >
                        No
                      </span>
                    </div>
                  </div>
                </p-selectbutton>
              </div>
              <div
                class="item"
              >
                <label>
                  Start Time
                </label>
                <p-calendar
                  class="ng-untouched ng-pristine ng-valid"
                  dateformat="yy/mm/dd"
                  formcontrolname="date"
                  hourformat="24"
                >
                  <span
                    class="p-calendar"
                  >
                    <input
                      autocomplete="off"
                      class="p-inputtext p-component"
                      placeholder=""
                      type="text"
                    />
                  </span>
                </p-calendar>
              </div>
              <div
                class="item"
              >
                <label>
                  Alarm Interval Min
                </label>
                <input
                  class="ng-untouched ng-pristine ng-valid"
                  formcontrolname="alarmIntervalMin"
                  placeholder="Numbers only"
                  type="number"
                />
              </div>
            </div>
          </form>
          <form
            class="frequencySetting ng-untouched ng-pristine ng-invalid"
            novalidate=""
          >
            <h4>
              Frequency Setting
            </h4>
            <div
              class="bg"
            >
              <div
                class="accordionGroup"
              >
                <div
                  class="accordionTool"
                >
                  <div
                    class="itemGroup"
                  >
                    <div
                      class="item"
                    >
                      <label
                        class="fieldRequired"
                      >
                        <span>
                          Last Tran Time
                        </span>
                      </label>
                    </div>
                    <div
                      class="item"
                    >
                      <label
                        class="fieldRequired"
                      >
                        <span>
                          Cron Expression
                        </span>
                      </label>
                    </div>
                    <div
                      class="item"
                    >
                      <button
                        class="txtBtn small customBtn add"
                      >
                         Generate Cron 
                      </button>
                    </div>
                  </div>
                  <div
                    class="btnGroup"
                  >
                    <button
                      class="iconBtn small customBtn add"
                    />
                  </div>
                </div>
                <div
                  class="accordionItem"
                >
                  <div
                    class="accordionHeader"
                  >
                    <div
                      class="itemGroup"
                    >
                      <div
                        class="item"
                      >
                        <p-calendar
                          class="ng-untouched ng-pristine ng-invalid"
                          dateformat="yy/mm/dd"
                          formcontrolname="LastTranTime"
                          hourformat="24"
                        >
                          <span
                            class="p-calendar"
                          >
                            <input
                              autocomplete="off"
                              class="p-inputtext p-component"
                              placeholder=""
                              type="text"
                            />
                          </span>
                        </p-calendar>
                      </div>
                      <div
                        class="item"
                      >
                        <input
                          class="ng-untouched ng-pristine ng-invalid"
                          formcontrolname="CronExpression"
                          placeholder="0 5 * ? * * *"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="btnGroup"
                    >
                      <button
                        class="iconBtn small customBtnReverseLine remove"
                      />
                      <button
                        class="iconBtn small customBtn copy"
                      />
                    </div>
                  </div>
                  <div
                    class="accordionContent"
                  >
                    <div
                      class="itemGroup"
                    >
                      <div
                        class="item"
                      >
                        <label>
                          Time Zone ID
                        </label>
                        <p-autocomplete
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="TimeZone"
                          placeholder="Select"
                        >
                          <span
                            class="p-autocomplete p-component p-autocomplete-dd"
                          >
                            <input
                              aria-activedescendant="p-highlighted-option"
                              aria-autocomplete="list"
                              aria-controls="pr_id_2_list"
                              aria-expanded="false"
                              aria-haspopup="true"
                              autocomplete="off"
                              class="p-autocomplete-input p-inputtext p-component p-autocomplete-dd-input"
                              placeholder="Select"
                              role="searchbox"
                              type="text"
                            />
                            <button
                              class="p-autocomplete-dropdown p-ripple p-button p-component p-button-icon-only"
                              pbutton=""
                              pripple=""
                              type="button"
                            >
                              <span
                                aria-hidden="true"
                                class="p-button-icon pi pi-chevron-down"
                              />
                              <span
                                aria-hidden="true"
                                class="p-button-label"
                              >
                              </span>
                            </button>
                          </span>
                        </p-autocomplete>
                      </div>
                      <div
                        class="item"
                      >
                        <label>
                          Retry Times
                        </label>
                        <input
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="RetryTimes"
                          placeholder="Numbers only"
                          type="number"
                        />
                      </div>
                      <div
                        class="item"
                      >
                        <label>
                          Loader Buffer Time
                        </label>
                        <input
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="LoaderBufferTime"
                          placeholder="Numbers only"
                          type="number"
                        />
                      </div>
                      <div
                        class="item"
                      >
                        <label>
                          Back To Buffer Time
                        </label>
                        <input
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="BackToBufferTime"
                          placeholder="Numbers only"
                          type="number"
                        />
                      </div>
                      <div
                        class="item"
                      >
                        <label>
                          Shift Back To Loader Time
                        </label>
                        <input
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="ShiftBackToLoaderTime"
                          placeholder="Numbers only"
                          type="number"
                        />
                      </div>
                    </div>
                    <div
                      class="itemGroup"
                    >
                      <div
                        class="item"
                      >
                        <label>
                          Start At
                        </label>
                        <p-calendar
                          class="ng-untouched ng-pristine ng-valid"
                          dateformat="yy/mm/dd"
                          formcontrolname="StartAt"
                        >
                          <span
                            class="p-calendar"
                          >
                            <input
                              autocomplete="off"
                              class="p-inputtext p-component"
                              placeholder=""
                              type="text"
                            />
                          </span>
                        </p-calendar>
                      </div>
                      <div
                        class="item"
                      >
                        <label>
                          End At
                        </label>
                        <p-calendar
                          class="ng-untouched ng-pristine ng-valid"
                          dateformat="yy/mm/dd"
                          formcontrolname="EndAt"
                        >
                          <span
                            class="p-calendar"
                          >
                            <input
                              autocomplete="off"
                              class="p-inputtext p-component"
                              placeholder=""
                              type="text"
                            />
                          </span>
                        </p-calendar>
                      </div>
                      <div
                        class="item"
                      >
                        <p>
                          Skip Over Due
                        </p>
                        <p-selectbutton
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="SkipOverDue"
                        >
                          <div
                            class="p-selectbutton p-buttonset p-component"
                            role="group"
                          >
                            <div
                              aria-label="Y"
                              aria-labelledby="Y"
                              aria-pressed="false"
                              class="p-button p-component p-ripple"
                              pripple=""
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="p-button-label"
                              >
                                Y
                              </span>
                            </div>
                            <div
                              aria-label="N"
                              aria-labelledby="N"
                              aria-pressed="true"
                              class="p-button p-component p-highlight p-ripple"
                              pripple=""
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="p-button-label"
                              >
                                N
                              </span>
                            </div>
                          </div>
                        </p-selectbutton>
                      </div>
                      <div
                        class="item"
                      >
                        <p>
                          Skip All Over Due
                        </p>
                        <p-selectbutton
                          class="ng-untouched ng-pristine ng-valid"
                          formcontrolname="SkipAllOverDue"
                        >
                          <div
                            class="p-selectbutton p-buttonset p-component"
                            role="group"
                          >
                            <div
                              aria-label="Y"
                              aria-labelledby="Y"
                              aria-pressed="false"
                              class="p-button p-component p-ripple"
                              pripple=""
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="p-button-label"
                              >
                                Y
                              </span>
                            </div>
                            <div
                              aria-label="N"
                              aria-labelledby="N"
                              aria-pressed="true"
                              class="p-button p-component p-highlight p-ripple"
                              pripple=""
                              role="button"
                              tabindex="0"
                            >
                              <span
                                class="p-button-label"
                              >
                                N
                              </span>
                            </div>
                          </div>
                        </p-selectbutton>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</app-txn-create>
`;
